### DISCUSSION TITLE
Shutdown with causing error.

### DISCUSSION THREAD

**Discussion by hsblhsn** [uber-go/fx#934] [Ideas]
The idea is mostly from https://github.com/uber-go/fx/issues/672 and my comment on that issue.

We can provide a way to log the error on exit with fxlog if any error provided with `ShutdownOption`.

What do you think?

**Comment by abhinav** [comment]
This sounds like a good idea.
We're already considering a `ShutdownCode(int)` option in #912,
so a `ShutdownCause(error)` that is then then returned by `app.Err()` makes sense.
